sdk/typescript/src/api/http-transport.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { RequestManager, HTTPTransport, Client } from '@open-rpc/client-js';
+import { PACKAGE_VERSION, TARGETED_RPC_VERSION } from '../version.js';
+import type { WebsocketClientOptions } from '../rpc/websocket-client.js';
+import { WebsocketClient } from '../rpc/websocket-client.js';
+
+/**
+ * An object defining headers to be passed to the RPC server
+ */
+export type HttpHeaders = { [header: string]: string };
+
+interface SuiHTTPTransportOptions {
+	network: SuiNetwork;
+	rpc?: {
+		headers?: HttpHeaders;
+		url?: string;
+	};
+	websocket?: WebsocketClientOptions & {
+		url?: string;
+	};
+}
+
+export interface SuiTransportRequestOptions {
+	method: string;
+	params: unknown[];
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type SuiNetwork =
```

Comment:
This is a little weird because fullnode URLs and networks are kind of separate concepts. It seems like this mostly exists to make transport construction easier, which I'm kind of fine if we don't over-optimize for.

Reply:
yeah, happy to rip this out if it doesn't make sense.  I think being able to get things working without extra imports is nice, but definitely open to changing this if it will become problematic in the future